[Python API] Added implicit input reshaping API and extended `reshape()` Unit Tests.

Jarvis2001 opened the pull request:
## Changes
- **New Overload**: Adds a new overload to [Model.reshape] that accepts a list of shapes (e.g., [model.reshape([[2,2], [1,3,224,244], [10]]). Each shape in the list is assigned to the corresponding input in order.

- **Simplified API**: This enables a more concise and intuitive way to reshape all model inputs, improving usability for common scenarios.

- **Tests**: Adds unit tests to ensure the new API works as expected, including error handling for mismatched input counts.
---
## Description

This PR also expands the Python API unit test suite for `Model.reshape()` to cover additional usage patterns, edge cases, and validation scenarios. It builds on the existing reshape tests by:

- Adding named input, port mapping, and `PartialShape` with tensor name coverage.  
- Testing layout-sensitive reshapes.  
- Verifying propagation of reshaped dimensions to all outputs.  
- Confirming reshape works on subgraphs constructed in-memory.  
- Ensuring invalid shapes trigger compile-time errors when appropriate.  
- Strengthening interval dimension tests to validate min/max bounds, dynamic status, and correct acceptance/rejection behaviour.

All new tests follow the same upstream style:

- In-memory model construction (no external IR files).  
- Minimal, deterministic setups.  
- Clear, direct assertions.  
- Compile step where needed to enforce runtime validation.

## Testing

All new tests pass locally against the latest OpenVINO build.


A clean version of: #31871

Closes: #18347

Jarvis2001 committed:
Done

Jarvis2001 committed:
Formatting

Jarvis2001 committed:
Formatting

Jarvis2001 committed:
Merge branch 'master' into variable-shape

almilosz commented on src/bindings/python/src/pyopenvino/graph/model.cpp:
```diff
@@ -489,6 +489,115 @@ void regclass_graph_Model(py::module m) {
+            py::dict shapes_dict = shapes.cast<py::dict>();
+            for (auto& item : shapes_dict) {
+                new_shapes.emplace(output_from_handle(self, item.first), partial_shape_from_handle(item.second));
+            }
```
You repeat here what [this](https://github.com/openvinotoolkit/openvino/blob/fab9a560f9b9bd06aa306be02a072ed644b0a61f/src/bindings/python/src/pyopenvino/graph/model.cpp#L662) reshape overload does

almilosz commented on src/bindings/python/tests/test_runtime/test_model.py:
```diff
@@ -594,20 +597,353 @@ def check_shape(new_shape):
+
+
+def test_dynamic_dimension_input_shape():
+    """Test reshaping a model with mixed dynamic and static dimensions.
```
I can't find `.reshape` method in this test

almilosz reviewed (changes requested):
Hello @Jarvis2001, 
I approved tests to run and left first round of review comments. Please fix them and pay attention to existing overloads of `Model.reshape`. Maybe you just have to change helper `Common::partial_shape_from_list`? 
Have a nice day, 
Alicja

Jarvis2001 committed:
Merge branch 'openvinotoolkit:master' into variable-shape

Jarvis2001 committed:
Modified tests, added helper functions

almilosz reviewed (changes requested):
Hello, 
there is still issue with linting. 
Please check that right now we have  `reshape(py::list)` i `reshape(list, optional variable_shapes)`

Jarvis2001 committed:
Merge branch 'openvinotoolkit:master' into variable-shape

Jarvis2001 committed:
Merge branch 'openvinotoolkit:master' into variable-shape

Jarvis2001 committed:
Fixed linter errors and added one test

Jarvis2001 committed:
-

Jarvis2001 committed:
Merge branch 'master' into variable-shape

p-wysocki commented:
As @almilosz said previously, please fix the failing CI checks, which are mostly linters. In case of further inactivity this PR will be closed.

Jarvis2001 committed:
No linter issues now I hope. The CI failed due to whitespace on line 557 in model.cpp

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
fixed the linter errors(line too long) in test_model.py

Jarvis2001 committed:
fixed linter E501 in helper.py

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

mlukasze committed:
Merge branch 'master' into variable-shape

Practice2001 committed:
Fix linter and CI issues

- Resolved previous Python linter errors
- Addressed new flake8 warnings (E501: line too long, E121/E126: indentation)
- Fixed copyright header issue in CI
- Verified build and tests locally

Jarvis2001 committed:
Reapply Practice2001 changes under Jarvis2001

Jarvis2001 committed:
Reapply Practice2001 changes under Jarvis2001

Jarvis2001 committed:
Applied changes suggested by copilot. Reformatted test on line 815 in test_model.py.

Jarvis2001 committed:
Revert "Fix linter and CI issues"

This reverts commit 446f1da4cb81290c78c7c1094ea89e626039131a.

Jarvis2001 committed:
Small changes

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Fix duplicate imports and lint issues in Python binding tests

Jarvis2001 committed:
Merge branch 'master' into variable-shape

Jarvis2001 committed:
Fixed Python multi-input reshape to preserve inputs by output identity, prevents duplicate or empty input names from overwriting shapes in the multi-input path.
Along with it a regression test covering duplicate input names.

Jarvis2001 commented:
Ping.
